hazelcast-code-generator/src/main/java/com/hazelcast/client/protocol/generator/CodecModel.java

```diff
@@ -0,0 +1,597 @@
+                setterString.append(getMapSetterString(type, name));
+            } else if (type.startsWith("java.util.List<") || type.startsWith("java.util.Set<")
+                    || type.startsWith("java.util.Collection<")) {
+                setterString.append(getCollectionSetterString(type, name));
+            } else if (type.endsWith("[]")) {
+                setterString.append(getArraySetterString(type, name));
+            } else {
+                setterString.append("clientMessage.set(" + name + ");");
+            }
+            return setterString.toString();
+        }
+
+        private String getArraySetterString(String type, String name) {
+            String itemVariableName = name + "_item";
+
+            String itemVariableType = CodeGenerationUtils.getTypeInsideData(type);
+
+            StringBuilder builder = new StringBuilder();
+            builder.append("clientMessage.set(" + name + ".length);\n        ");
+            builder.append("for (" + itemVariableType + " " + itemVariableName + " : " + name + " ) {\n        ");
+            builder.append("    " + resolveDataSetterStringJava(itemVariableType, itemVariableName) + "\n        ");
+            builder.append("}\n        ");
+            return builder.toString();
+        }
+
+        private String getCollectionSetterString(String type, String name) {
+            String itemType = CodeGenerationUtils.getTypeInsideCollection(type);
+            String itemVariableName = name + "_item";
+
+            StringBuilder builder = new StringBuilder();
+            builder.append("clientMessage.set(" + name + ".size());\n        ");
+            builder.append("for (" + itemType + " " + itemVariableName + ": " + name + " ) {\n        ");
+            builder.append("    " + resolveDataSetterStringJava(itemType, itemVariableName) + "\n        ");
+            builder.append("}\n        ");
+            return builder.toString();
+        }
+
+        private String getMapSetterString(String type, String name) {
+            StringBuilder builder = new StringBuilder();
+
+            String keyType = CodeGenerationUtils.getKeyTypeInsideMap(type);
+            builder.append("java.util.Set<" + keyType + "> " + name
+                    + "_keySet = (java.util.Set<" + keyType + ">) " + name + ".keySet();\n     ");
+            builder.append(resolveDataSetterStringJava("java.util.Set<" + keyType + "> ", name + "_keySet"));
+
+            String valueType = CodeGenerationUtils.getValueTypeInsideMap(type);
+            builder.append("java.util.List<" + valueType + "> " + name
+                    + "_values = (java.util.List<" + valueType + "> )" + name + ".values();\n       ");
+            builder.append(resolveDataSetterStringJava("java.util.List<" + valueType + ">", name + "_values"));
+
+            return builder.toString();
+        }
+//
+//        public String getDataSetterStringCSharp() {
+//            String getterString;
```

Comment:
This part is also required for c# generator but it will be enabled during c# dev